pkg/storegateway/bucket_test.go

```diff
@@ -1929,6 +1930,125 @@ func TestSeries_BlockWithMultipleChunks(t *testing.T) {
 	}
 }
 
+func TestBucketStore_Series_LimitsWithStreamingEnabled(t *testing.T) {
+	var (
+		ctx    = context.Background()
+		tmpDir = t.TempDir()
+		bktDir = filepath.Join(tmpDir, "bucket")
+		logger = log.NewNopLogger()
+	)
+
+	const (
+		numSamplesPerSeries = 10 // A low number so that all samples fit in a single chunk.
+		minTime             = 0
+		maxTime             = 1000
+	)
+
+	// Create two blocks. Some series exists in both blocks, some don't.
+	// Samples for the overlapping series are equal between the two blocks
+	// (simulate the case of uncompacted blocks from ingesters).
+	_, err := testhelper.CreateBlock(ctx, bktDir, []labels.Labels{
+		labels.FromStrings(labels.MetricName, "series_1"),
+		labels.FromStrings(labels.MetricName, "series_2"),
+	}, numSamplesPerSeries, minTime, maxTime, nil, 0)
+	require.NoError(t, err)
+
+	_, err = testhelper.CreateBlock(ctx, bktDir, []labels.Labels{
+		labels.FromStrings(labels.MetricName, "series_1"),
+		labels.FromStrings(labels.MetricName, "series_2"),
+	}, numSamplesPerSeries, minTime, maxTime, nil, 0)
+	require.NoError(t, err)
+
+	// Create a bucket and upload the block there.
+	bkt, err := filesystem.NewBucket(bktDir)
+	assert.NoError(t, err)
+	defer func() { assert.NoError(t, bkt.Close()) }()
+
+	instrBkt := objstore.WithNoopInstr(bkt)
+
+	// Instance a real bucket store we'll use to query the series.
+	fetcher, err := block.NewMetaFetcher(logger, 10, instrBkt, tmpDir, nil, nil)
+	assert.NoError(t, err)
+
+	tests := map[string]struct {
+		reqMatchers    []storepb.LabelMatcher
+		seriesLimit    uint64
+		chunksLimit    uint64
+		expectedErr    string
+		expectedSeries int
+	}{
+		"should fail if the number of unique series queried is less than the configured series limit": {
```

Comment:
I am a bit confused with the test title: from what I understood the test should fail if the number of unique series queried is greater than the configured series limit. 
For example, the actual error obtained by executing this test is: "expand series set: exceeded series limit: limit 1 exceeded", which is the opposite of what the test title says.
If I am wrong, then I need to understand the purpose of the limiter better. If I am right, then other test titles should also be updated.

Reply:
You're right. Should be changed to "greater than". Will fix, thanks!

Reply:
See: https://github.com/grafana/mimir/pull/4051